### PR TITLE
[#3027] carousel template bug

### LINKED ISSUE #3027: Carousel Template bug
**Current state:**  the carousels are buggy
- the arrows and navigation don't work well 

Seen on cloud instances ([cloud #302](https://github.com/airyhq/cloud/issues/302))

### LINKED ISSUE #3035: Google message bug
- Google suggestions are broken in case of multiple suggestions 

Screenshot 2022-04-14 at 16 27 51

### LINKED ISSUE #3045: Resend failed message improvement
Right now, when a user resends a failed message: 
- there is no feedback from the UI, which gives the impression that the feature is not working
- the MessageList scrolls to the bottom after the resent, which looks buggy 

**To do:**

When the user clicks on 'Retry':

- [x] - remove the failed icon and text while the request goes through 
- [x] - if the status is still failed after the request, display the failed icon and text 
- [x] - scroll to the message that was resent so that the user can see it

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #3027 
closes #3035 
closes #3045

### CODE REVIEW

File: frontend/ui/src/pages/Inbox/Messenger/MessageList/index.tsx [hunk 1]
```diff
@@ -132,8 +139,9 @@ const MessageList = (props: MessageListProps) => {
...
   };
 
-  const handleFailedMessage = (resend: boolean, messageId: string) => {
-    resend && resendMessage({messageId});
+  const handleFailedMessage = (messageId: string) => {
+    resendMessage({messageId});
+    setResentMessage(true);
   };
 
   return (
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Inbox/Messenger/MessageList/index.tsx:147]: This prevents to scroll to the bottom when a failed state message is resent

File: lib/typescript/components/general/Carousel/index.tsx [hunk 2]
```diff
@@ -74,6 +78,7 @@ export const Carousel = ({children}) => {
           carouselChildren.current.scrollLeft + getScrollBy(),
           maximumScrollRight(carouselChildren.current)
         ),
+
         behavior: 'smooth',
       });
     }, 1000),
```
Comment (AudreyKj) [inline lib/typescript/components/general/Carousel/index.tsx:84]: carousel bug fix

File: lib/typescript/components/message/MessageInfo/index.tsx [hunk 3]
```diff
@@ -31,16 +41,31 @@ const MessageSuccessfull = ({sentAt, senderName, senderIdentity}) => {
...
+            setFailedMessageResent={setFailedMessageResent}
+          />
         ) : (
-          <MessageSuccessfull sentAt={sentAt} senderName={senderName} senderIdentity={senderIdentity} />
+          !failedMessageResent && (
+            <MessageSuccessfull sentAt={sentAt} senderName={senderName} senderIdentity={senderIdentity} />
+          )
         )}
       </span>
     </div>
```
Comment (AudreyKj) [inline lib/typescript/components/message/MessageInfo/index.tsx:71]: failed message improvement

**Review by thorstenairy — APPROVED** [review]
LGTM

**Pull request merged**